### DISCUSSION TITLE
How to write an asset that depends on itself

### DISCUSSION THREAD

**Discussion by meyergin** [dagster-io/dagster#15658] [Q&A] [answered]
how to write an asset which get the input of object stored by itself
scenario: get the last processed data and update/append with new fetched data

**Comment by tacastillo** [comment, marked as answer]
Hi! Here's a GitHub Discussion we have with a couple options, depending on your setup:

https://github.com/dagster-io/dagster/discussions/14432

**Reply by meyergin** [reply]
Got that, it works though not so elegant.
Thanks for your help.
